src/paginationCalls/pageDescriptor.ts

```diff
@@ -156,7 +156,14 @@ export class PageDescriptor implements Descriptor {
                 nextPageRequest!,
                 options
               )) as ResultTuple;
-              cache.push(...(result as ResponseType[]));
+              // For pagination response with protobuf map type, use tuple as representation.
+              if (result && !Array.isArray(result)) {
+                for (const [key, value] of Object.entries(result)) {
+                  cache.push([key, value]);
```

Comment:
I am not sure if we need to define a type for this tuple. @alexander-fenster what do you suggest the type here? My guess here is that we could define a MapResponseTuple
```
type MapResponseTuple = [String, ??(not sure)]
```

Reply:
It would be `[string, ResponseType]`, and the `cache` should be defined as `ResponseType | [string, ResponseType]`.

Reply:
(I think having just a `ResponseType` which is basically any object is OK)